corehq/motech/repeaters/tests/test_repeater.py

```diff
@@ -385,20 +385,10 @@ def test_retry_process_repeat_record_locking(self):
             self.assertEqual(mock_process.delay.call_count, 2)
 
     def test_automatic_cancel_repeat_record(self):
-        instance_id = uuid.uuid4().hex
-        case_block = CaseBlock(
-            case_id=CASE_ID,
-            create=False,
-            case_name='ABC 345',
-        ).as_text()
-        xform_xml = XFORM_XML_TEMPLATE.format(
-            'https://www.commcarehq.org/test/repeater/',
-            USER_ID,
-            instance_id,
-            case_block,
-        )
-        self.post_xml(xform_xml, self.domain)
-        repeat_record = RepeatRecord.objects.get(payload_id=instance_id)
+        case = CommCareCase.objects.get_case(CASE_ID, self.domain)
+        rr = self.case_repeater.register(case)
+        # Fetch the revision that was updated:
+        repeat_record = RepeatRecord.objects.get(id=rr.id)
```

Comment:
This reverts the change made in https://github.com/dimagi/commcare-hq/pull/35734/commits/1659c5e72c6a2457e221ae30dd04d528a0773455 which was necessary because `CaseRepeater.register()` could return `None` instead of `RepeatRecord` (line 389).

That behavior was not a good idea; `CaseRepeater.register()` should maintain the return value of `Repeater.register()`.